util/db/db.go

```diff
@@ -41,6 +42,9 @@ type ArgoDB interface {
 	CreateRepoCertificate(ctx context.Context, certificate *appv1.RepositoryCertificateList, upsert bool) (*appv1.RepositoryCertificateList, error)
 	// CreateRepoCertificate creates a new certificate entry
 	RemoveRepoCertificates(ctx context.Context, selector *CertificateListSelector) (*appv1.RepositoryCertificateList, error)
+
+	// ListHelmRepositories lists repositories
+	ListHelmRepositories(ctx context.Context) ([]*appv1.Repository, error)
```

Comment:
// DERECATED

Reply:
This one is not really deprecated. Even after we remove legacy helm repos it still makes sense to use this method to get only repos with helm type. In one of our Argo CD instances, we have ~350 git repos and 0 helm repos. So it does not makes sense to send 350 repos to Repo Server, instead it is better to filter out non-helm repos on the client side.